ppsci/experimental/math_module.py

```diff
@@ -462,3 +463,183 @@ def trapezoid_integrate(
+
+        Returns:
+            backend tensor: Quadrature result
+        """
+        scales = integration_domain[:, 1] - integration_domain[:, 0]
+        volume = paddle.prod(scales)
+
+        # Integral = V / N * sum(func values)
+        N = function_values.shape[0]
+        integral = volume * paddle.sum(function_values, axis=0) / N
+        return integral
+
+    def calculate_sample_points(
+        N: int, integration_domain: paddle.Tensor, seed: int = None
+    ):
+        """Calculate random points for the integrand evaluation
+
+        Args:
+            N (int): Number of points
+            integration_domain (backend tensor): Integration domain
+            seed (int, optional): Random number generation seed for the sampling point creation, only set if provided. Defaults to None.
+        Returns:
+            Sample points
+        """
+        dim = integration_domain.shape[0]
+        domain_starts = integration_domain[:, 0]
+        domain_sizes = integration_domain[:, 1] - domain_starts
+        # Scale and translate random numbers via broadcasting
+        return (
+            paddle.uniform(
+                shape=[N, dim],
+                dtype=domain_sizes.dtype,
+                min=0.0,
+                max=1.0,
+                seed=seed or 0,
+            )
+            * domain_sizes
+            + domain_starts
+        )
+
+    if dim is not None:
+        if dim < 1:
+            raise ValueError("Dimension needs to be 1 or larger.")
+        if N is not None:
+            if N < 1 or type(N) is not int:
+                raise ValueError("N has to be a positive integer.")
+
+    integration_domain = _setup_integration_domain(dim, integration_domain)
+    sample_points = calculate_sample_points(N, integration_domain, seed)
+    function_values, _ = evaluate_integrand(fn, sample_points)
+    return calculate_result(function_values, integration_domain)
+
+
+def _setup_integration_domain(
+    dim: int, integration_domain: Optional[List[List[float]], paddle.Tensor]
+) -> paddle.Tensor:
+    """Sets up the integration domain if unspecified by the user.
+    Args:
+        dim (int): Dimensionality of the integration domain.
+        integration_domain (List or Tensor): Integration domain, e.g. [[-1,1],[0,1]]. Defaults to [-1,1]^dim.
+    Returns:
+        Integration domain.
+    """
+    # If no integration_domain is specified, create [-1,1]^d bounds
+    if integration_domain is None:
+        integration_domain = [[-1.0, 1.0]] * dim
+
+    integration_domain = [[float(b) for b in bounds] for bounds in integration_domain]
+
+    integration_domain = paddle.to_tensor(integration_domain)
+
+    if tuple(integration_domain.shape) != (dim, 2):
+        raise ValueError(
+            "The integration domain has an unexpected shape. "
+            f"Expected {(dim, 2)}, got {integration_domain.shape}"
+        )
+    return integration_domain
+
+
+def evaluate_integrand(fn, points, weights=None, args=None):
```

Comment:
该函数为非公开的辅助函数，前面加上单个下划线
```suggestion
def _evaluate_integrand(fn, points, weights=None, args=None):
```

Reply:
DONE